### PR TITLE
(DOCSP-31533): RN migrate PBS to FS updates

### PR DESCRIPTION
**Pull request by dacharyc**
### Jira

- https://jira.mongodb.org/browse/DOCSP-31533

### Staged Changes

- [Sync Data](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-31533/sdk/react-native/sync-data/): Remove PBS, add tip about PBS being an older sync mode and recommending new apps use Flexible Sync
- [Partition-Based Sync](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-31533/sdk/react-native/sync-data/partition-based-sync/): Add Migration info to page, move Progress Notification docs from Manage a Sync Session to this page since they only work with PBS right now
- [Manage a Sync Session](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-31533/sdk/react-native/sync-data/manage-sync-session/): Remove Progress Notification documentation from the page since it only works with PBS

### Reminder Checklist

If your PR modifies the docs, you might need to also update some corresponding
pages. Check if completed or N/A.

- [x] Create Jira ticket for corresponding docs-app-services update(s), if any
- [x] Checked/updated Admin API
- [x] Checked/updated CLI reference

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sdk/react-native/sync-data.txt [hunk 1]
```diff
@@ -81,20 +81,10 @@ Group Updates for Improved Performance
 
 .. include:: /includes/sync-memory-performance.rst
 
-.. _react-native-partition-based-sync-fundamentals:
+.. tip::
```
Comment (cbullinger) [inline source/sdk/react-native/sync-data.txt:84]: Can't comment on it, but there are still mentions of PBS at top of this page (e.g. lines 30-38)

File: source/sdk/react-native/sync-data/partition-based-sync.txt [hunk 2]
```diff
@@ -18,6 +18,20 @@ For more information about Partition-Based Sync and how to configure it
 in Atlas App Services, refer to :ref:`Partition-Based Sync <partition-based-sync>`
 in the App Services documentation.
 
+.. _react-native-partition-based-sync-fundamentals:
```
Comment (cbullinger) [inline source/sdk/react-native/sync-data/partition-based-sync.txt:21]: Can't comment on it, but should "Flexible Sync" in line 14 be a link to the `react-native/sync-data` page?

**Review by cbullinger — APPROVED** [review]
Should we also move the PBS page to the bottom of the TOC for consistency?

File: source/sdk/react-native/sync-data.txt (hunk 1 above)
Comment (dacharyc) [reply source/sdk/react-native/sync-data.txt:84]: Ahh, thanks for the catch, I missed that! Removing.

File: source/sdk/react-native/sync-data/partition-based-sync.txt (hunk 2 above)
Comment (dacharyc) [reply source/sdk/react-native/sync-data/partition-based-sync.txt:21]: Makes sense, adding it!

**Pull request merged**